lib/screen/quiz_screen.dart

```diff
@@ -86,114 +97,152 @@ class _QuizScreenState extends State<QuizScreen> with SingleTickerProviderStateM
+                                      _selectedAnswers.add(answer);
+                                    }
+                                  } else {
+                                    _selectedAnswers.clear();
+                                    _selectedAnswers.add(answer);
+                                  }
+                                });
+                              },
                               child: Container(
                                 margin: EdgeInsets.symmetric(vertical: 8),
                                 padding: EdgeInsets.symmetric(horizontal: 12, vertical: 8),
                                 decoration: BoxDecoration(
+                                  color: isSelected ? Color(0xFF098EAB).withOpacity(0.1) : Colors.transparent,
                                   border: Border.all(
-                                    color: isSelected ? Colors.green : Color(0xFF098EAB),
+                                    color: isSelected ? Color(0xFF098EAB) : Colors.grey,
                                     width: 2,
                                   ),
                                   borderRadius: BorderRadius.circular(8),
                                 ),
-                                child: RadioListTile<Answer>(
-                                  title: Text(answer.answerText, style: TextStyle(fontSize: 18, color: Colors.black)),
-                                  value: answer,
-                                  groupValue: _selectedAnswer,
-                                  contentPadding: EdgeInsets.zero,
-                                  activeColor: Color(0xFF098EAB),
-                                  onChanged: (Answer? value) {
-                                    setState(() {
-                                      _selectedAnswer = value;
-                                    });
-                                  },
+                                child: Row(
+                                  children: [
+                                    if (_quizCategory!.questions[_currentQuestionIndex].isIndexBased)
+                                      Checkbox(
+                                        value: isSelected,
+                                        activeColor: Color(0xFF098EAB),
+                                        onChanged: (bool? value) {
+                                          setState(() {
+                                            if (value == true) {
+                                              _selectedAnswers.add(answer);
+                                            } else {
+                                              _selectedAnswers.remove(answer);
+                                            }
+                                          });
+                                        },
+                                      ),
+                                    if (!_quizCategory!.questions[_currentQuestionIndex].isIndexBased)
```

Comment:
Use _quizCategory?.questions[_currentQuestionIndex].isIndexBased == true instead of null operator